plutus-core/plutus-ir/src/PlutusIR/Compiler.hs

```diff
@@ -40,21 +46,50 @@ import qualified PlutusIR.Transform.Unwrap          as Unwrap
 import           PlutusIR.TypeCheck.Internal
 
 import qualified PlutusCore                         as PLC
-import qualified PlutusCore.Constant.Meaning        as M
 
+import           Control.Lens
 import           Control.Monad
 import           Control.Monad.Reader
 import           PlutusPrelude
 
 -- | Actual simplifier
 simplify
-    :: (M.ToBuiltinMeaning uni fun, PLC.MonadQuote m)
+    :: forall m e uni fun a b. Compiling m e uni fun a
     => Term TyName Name uni fun b -> m (Term TyName Name uni fun b)
-simplify = DeadCode.removeDeadBindings <=< Inline.inline . Beta.beta . Unwrap.unwrapCancel
+simplify term =
+  do
+    selectedPasses <- map snd <$> filterM (shouldRunPass . fst) availablePasses
```

Comment:
I really like `witherable` for writing more "pointful" versions of things like this, and we even have a dependency on it already. Something like:
```
selectedPasses <- forMaybe availablePasses $ \(optionAccessor, pass) -> do
    shouldRun <- shouldRunPass optionAccessor
    if shouldRun then pure $ Just pass else pure Nothing
```

Reply:
Alternatively, you already wrote `runIf`, you could also map `uncurry runIf` over the list, turning the non-running passes into no-ops!

Reply:
@michaelpj I moved those changes to a separate PR:
https://github.com/input-output-hk/plutus/pull/3655

I was thinking to implement it more like:
https://github.com/input-output-hk/plutus/blob/46abe903143e092a0877586d901b1afe9d5a12db/plutus-core/plutus-ir/src/PlutusIR/Compiler.hs#L60-L106